### PR TITLE
Breadth first search parsed literal

### PR DESCRIPTION
**Pull request by nitishchauhan0022**
Fixes #2244 .

Changes proposed in this pull request:
- doc/breadthFirstSearch/pgr_breadthFirstSearch.rst
- doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst

@pgRouting/admins

### CODE REVIEW

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst [hunk 1]
```diff
@@ -14,16 +14,13 @@
   (`3.3 <https://docs.pgrouting.org/3.3/en/pgr_binaryBreadthFirstSearch.html>`__)
   `3.2 <https://docs.pgrouting.org/3.2/en/pgr_binaryBreadthFirstSearch.html>`__
 
-pgr_binaryBreadthFirstSearch - Experimental
+``pgr_binaryBreadthFirstSearch - Experimental``
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:17]: Only the name of the function has the double back tick

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst [hunk 2]
```diff
@@ -14,16 +14,13 @@
...
 ===============================================================================
 
 ``pgr_binaryBreadthFirstSearch`` — Returns the shortest path(s) in a binary graph.
-Any graph whose edge-weights belongs to the set {0,X}, where 'X' is any non-negative real integer, is termed as a 'binary graph'.
+Any graph whose edge-weights belongs to the set {0,X}, where 'X' is any 
+non-negative real integer, is termed as a 'binary graph'.
 
-.. figure:: images/boost-inside.jpeg
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:23]: Do not remove the references to the boost functions

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst [hunk 3]
```diff
@@ -33,75 +30,82 @@ Any graph whose edge-weights belongs to the set {0,X}, where 'X' is any non-nega
...
-However if the weights are more constrained, we can use a faster algorithm. This algorithm, termed as 'Binary Breadth First Search' as well as '0-1 BFS', is a variation of the standard Breadth First Search problem to solve the SSSP (single-source shortest path) problem in :math:`O(|E|)`, if the weights of each edge belongs to the set {0,X}, where 'X' is any non-negative real integer.
+However if the weights are more constrained, we can use a faster algorithm.
+This algorithm, termed as 'Binary Breadth First Search' as well as '0-1 BFS',
+is a variation of the standard Breadth First Search problem to solve the 
+SSSP (single-source shortest path) problem in :math:`O(|E|)`, if the weights
+of each edge belongs to the set {0,X}, where 'X' is any non-negative real integer.
 
 
 **The main Characteristics are:**
+  * Process is done only on 'binary graphs'. ('Binary Graph': Any graph whose
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:65]: A list starts with the asterisk in the first column of the line

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst [hunk 4]
```diff
@@ -33,75 +30,82 @@ Any graph whose edge-weights belongs to the set {0,X}, where 'X' is any non-nega
...
+    pgr_binaryBreadthFirstSearch(`Edges SQL`_, **start vid**, **end vid** [, directed]);
     RETURNS SET OF (seq, path_seq, node, edge, cost, agg_cost)
     OR EMPTY SET
 
-:Example: From vertex :math:`2` to vertex  :math:`3` on an **undirected** binary graph
+:Example: From vertex :math:`2` to vertex  :math:`3` on an **directed** binary graph
 
 .. literalinclude:: doc-pgr_binaryBreadthFirstSearch.queries
-   :start-after: -- q2
+   :start-after: -- q1
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:108]: Generate the documentation locally and verify what query is between q1 and q3 
Should be only one query.

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst [hunk 5]
```diff
@@ -221,11 +228,9 @@ Return Columns
     :end-before: return_path_end
 
 
-Example Data
+Additional Examples
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:231]: Now that you are on it, use the sample data of the documentation.
To do that you need to use `edge_table`.
You will have to modify docqueries/breadthFirstSearch/doc-pgr_binaryBreadthFirstSearch.test.sql
Lets leave this for a next PR. (aka do not do it on this one)
Mention in the #2244 issue that pgr_binaryBreadthFirstSearch needs a revision of the examples used

File: doc/breadthFirstSearch/pgr_breadthFirstSearch.rst [hunk 6]
```diff
@@ -16,10 +16,11 @@
   `3.1 <https://docs.pgrouting.org/3.1/en/pgr_breadthFirstSearch.html>`__
   `3.0 <https://docs.pgrouting.org/3.0/en/pgr_breadthFirstSearch.html>`__
 
-pgr_breadthFirstSearch - Experimental
+``pgr_breadthFirstSearch - Experimental``
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_breadthFirstSearch.rst:19]: double tick only on the function name

File: doc/breadthFirstSearch/pgr_breadthFirstSearch.rst [hunk 7]
```diff
@@ -34,26 +35,31 @@ pgr_breadthFirstSearch - Experimental
...
 Signatures
 -------------------------------------------------------------------------------
 
-.. code-block:: none
+.. parsed-literal:: 
 
-    pgr_breadthFirstSearch(Edges SQL, Root vid [, max_depth] [, directed])
-    pgr_breadthFirstSearch(Edges SQL, Root vids [, max_depth] [, directed])
+    pgr_breadthFirstSearch(`Edges SQL`_, **Root vid** [, max_depth] [, directed])
+    pgr_breadthFirstSearch(`Edges SQL`_, **Root vids** [, max_depth] [, directed])
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_breadthFirstSearch.rst:63]: Remove the blank line before RETURNS

File: doc/breadthFirstSearch/pgr_breadthFirstSearch.rst [hunk 8]
```diff
@@ -63,13 +69,13 @@ Signatures
 Single Vertex
 ...............................................................................
 
-.. code-block:: none
+.. parsed-literal::
 
-    pgr_breadthFirstSearch(Edges SQL, Root vid [, max_depth] [, directed])
+    pgr_breadthFirstSearch(`Edges SQL`_, **Root vid** [, max_depth] [, directed])
 
     RETURNS SET OF (seq, depth, start_vid, node, edge, cost, agg_cost)
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_breadthFirstSearch.rst:76]: Remove the blank line before RETURNS

File: doc/breadthFirstSearch/pgr_breadthFirstSearch.rst [hunk 9]
```diff
@@ -81,9 +87,9 @@ Single Vertex
 Multiple Vertices
 ...............................................................................
 
-.. code-block:: none
+.. parsed-literal:: 
 
-    pgr_breadthFirstSearch(Edges SQL, Root vids [, max_depth] [, directed])
+    pgr_breadthFirstSearch(`Edges SQL`_, **Root vids** [, max_depth] [, directed])
 
     RETURNS SET OF (seq, depth, start_vid, node, edge, cost, agg_cost)
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_breadthFirstSearch.rst:94]: Remove the blank line before RETURNS

File: doc/breadthFirstSearch/pgr_breadthFirstSearch.rst [hunk 10]
```diff
@@ -175,7 +182,8 @@ Additional Examples
 
 **Undirected Graph**
 
-:Example: The Breadth First Search traverls starting on vertices :math:`\{11, 12\}` with :math:`depth <= 2` as well as considering the graph to be undirected.
+**Example:** The Breadth First Search traverls starting on vertices :math:`\{11, 12\}`
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_breadthFirstSearch.rst:185]: Use:  
:Example:

File: doc/breadthFirstSearch/pgr_breadthFirstSearch.rst [hunk 11]
```diff
@@ -184,7 +192,8 @@ Additional Examples
 
 **Vertex Out Of Graph**
 
-:Example: The output of the function when a vertex not present in the graph is passed as a parameter.
+**Example:** The output of the function when a vertex not present in the graph
```
Comment (cvvergara) [inline doc/breadthFirstSearch/pgr_breadthFirstSearch.rst:195]: same as above

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst (hunk 4 above)
Comment (cvvergara) [reply doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:108]: Never mind I see you modified the file

File: docqueries/breadthFirstSearch/doc-pgr_binaryBreadthFirstSearch.result [hunk 12]
```diff
@@ -38,34 +38,126 @@ INSERT 0 18
...
+(5 rows)
+
+/* -- q6 */
 SELECT * FROM pgr_binaryBreadthFirstSearch(
-    'SELECT id, source, target, road_work as cost, reverse_road_work as reverse_cost FROM roadworks',
-    2, 3,
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table',
+    'SELECT * FROM combinations_table',
     FALSE
```
Comment (cvvergara) [inline docqueries/breadthFirstSearch/doc-pgr_binaryBreadthFirstSearch.result:126]: Use lower case

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst (hunk 5 above)
Comment (cvvergara) [reply doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:231]: Never mind I see you already did something about it

File: docqueries/breadthFirstSearch/doc-pgr_binaryBreadthFirstSearch.test.sql [hunk 13]
```diff
@@ -36,36 +36,60 @@ INSERT INTO roadworks(
...
+    FROM edge_table',
     ARRAY[2,11], 5
 );
 /* -- q5 */
 SELECT * FROM pgr_binaryBreadthFirstSearch(
-    'SELECT id, source, target, road_work as cost, reverse_road_work as reverse_cost FROM roadworks',
+    'SELECT id, source, target, cost, reverse_cost 
+    FROM edge_table',
     ARRAY[2,11], ARRAY[3,5],
     FALSE
```
Comment (cvvergara) [inline docqueries/breadthFirstSearch/doc-pgr_binaryBreadthFirstSearch.test.sql:61]: FALSE -> false

File: docqueries/breadthFirstSearch/doc-pgr_breadthFirstSearch.test.sql [hunk 14]
```diff
@@ -1,21 +1,25 @@
...
+    'SELECT id, source, target, cost, reverse_cost 
+    FROM edge_table ORDER BY id',
     ARRAY[11,12], max_depth := 2
 );
 /* --q3 */
 SELECT * FROM pgr_breadthFirstSearch(
-    'SELECT id, source, target, cost, reverse_cost FROM edge_table ORDER BY id',
+    'SELECT id, source, target, cost, reverse_cost 
+    FROM edge_table ORDER BY id',
     ARRAY[11,12], max_depth := 2, directed := false
```
Comment (cvvergara) [inline docqueries/breadthFirstSearch/doc-pgr_breadthFirstSearch.test.sql:17]: The latest syntax use => instead of :=

**Review by cvvergara — CHANGES_REQUESTED** [review]

**Comment by cvvergara** [comment]
When done with the changes please click on "Ready for review"

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst (hunk 2 above)
Comment (nitishchauhan0022) [reply doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:23]: Hi @cvvergara I am not sure where we need to use the boost graph inside reference. Actually, i referred the documentation of [pgr_edwardmoore](https://docs.pgrouting.org/latest/en/pgr_edwardMoore.html) and [pgr_chineasePostman](https://docs.pgrouting.org/latest/en/pgr_chinesePostman.html) . These docs are also not using the boost graph inside reference. So, I checked ther header files from include folder and I found that there is no direct implementation of these algorithm by boost. Similar case is for pgr_binaryBreadthFirstSearch

File: docqueries/breadthFirstSearch/doc-pgr_binaryBreadthFirstSearch.result (hunk 12 above)
Comment (nitishchauhan0022) [reply docqueries/breadthFirstSearch/doc-pgr_binaryBreadthFirstSearch.result:126]: @cvvergara  only for inner queries

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst (hunk 2 above)
Comment (cvvergara) [reply doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:23]: Did you read the C++ code of the function?

**Review by cvvergara — APPROVED** [review]

File: doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst (hunk 2 above)
Comment (nitishchauhan0022) [reply doc/breadthFirstSearch/pgr_binaryBreadthFirstSearch.rst:23]: Yes mam, I have gone through the pgr_binaryBreadthFirstSearch.hpp file.
In other shortest path finding algorithm implementations, the private section of the main function class in header file consists of  call to the boost graph algorithm which is not the case for pgr_binaryBreadthFirstSearch.hpp It consists of local implementation.

**Pull request merged**